bitcoin/src/sign_message.rs

```diff
@@ -159,37 +157,40 @@ mod message_signing {
                 None => Ok(false),
             }
         }
-
-        /// Convert a signature from base64 encoding.
-        #[cfg(feature = "base64")]
-        pub fn from_base64(s: &str) -> Result<MessageSignature, MessageSignatureError> {
-            let bytes = base64::decode(s).map_err(|_| MessageSignatureError::InvalidBase64)?;
-            MessageSignature::from_slice(&bytes)
-        }
-
-        /// Convert to base64 encoding.
-        #[cfg(feature = "base64")]
-        pub fn to_base64(self) -> String { base64::encode(&self.serialize()[..]) }
     }
 
     #[cfg(feature = "base64")]
-    impl fmt::Display for MessageSignature {
-        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-            let bytes = self.serialize();
-            // This avoids the allocation of a String.
-            write!(
-                f,
-                "{}",
-                base64::display::Base64Display::with_config(&bytes[..], base64::STANDARD)
-            )
+    mod base64_impls {
+        use base64::prelude::{Engine as _, BASE64_STANDARD as base64_std};
+
+        use super::*;
+        use crate::prelude::String;
+
+        impl MessageSignature {
+            /// Convert a signature from base64 encoding.
+            pub fn from_base64(s: &str) -> Result<MessageSignature, MessageSignatureError> {
+                let bytes =
+                    base64_std.decode(s).map_err(|_| MessageSignatureError::InvalidBase64)?;
+                MessageSignature::from_slice(&bytes)
+            }
+
+            /// Convert to base64 encoding.
+            pub fn to_base64(self) -> String { base64_std.encode(&self.serialize()[..]) }
         }
-    }
 
-    #[cfg(feature = "base64")]
-    impl core::str::FromStr for MessageSignature {
-        type Err = MessageSignatureError;
-        fn from_str(s: &str) -> Result<MessageSignature, MessageSignatureError> {
-            MessageSignature::from_base64(s)
+        impl fmt::Display for MessageSignature {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                let bytes = self.serialize();
+                // This avoids the allocation of a String.
+                write!(f, "{}", base64::display::Base64Display::new(&bytes[..], &base64_std))
```

Comment:
```suggestion
                write!(f, "{}", base64::display::Base64Display::new(&bytes, &base64_std))
```

We don't need the explicit slice thing anymore now we have MSRV 1.48, we did for Rust 1.41